back/engines/free/email_campaigns/app/models/email_campaigns/campaign.rb

```diff
@@ -41,9 +43,17 @@ class Campaign < ApplicationRecord
 
     before_validation :set_enabled, on: :create
 
+    validates :context_id, absence: true, unless: :skip_context_absence?
```

Comment:
Maybe we can get rid of `skip_context_absence?` if we are confident it only applies to manual campaigns:
```suggestion
    validates_presence_of :context_id, if: :manual?
    validates_absence_of :context_id, unless: :manual?
```

or something like:
```suggestion
    validate :validate_context_id_presence

    def validate_context_id_presence
      if manual?
        errors.add(:context_id, :blank) if context_id.blank?
      else
        errors.add(:context_id, :present) if context_id.present?
      end
    end
```

Reply:
It only applies to the new campaign, not the existing `Manual` campaign. So, `if: manual?` is not sufficient.